File: events/get_related_data_for_event.py
# this Python code example gets related profile and metric data for a given event.
import os
from klaviyo_api import KlaviyoAPI

# load your private API key from environment variables. 
# ensure you have set the environment variable "private_api_key" with Klaviyo private API key that has read access to events, profiles, and metrics. 
private_api_key = os.environ.get("private_api_key")

# instantiate the KlaviyoAPI client with your private API key.
klaviyo = KlaviyoAPI(private_api_key)

# example 1: get related profile data for a specific event. Replace event_id with the ID of the event you are interested in.
# note that this is the event's Activity ID rather than Unique ID.
event_id = '6e3MwdH2fWm'

raw_profile_data = klaviyo.Events.get_profile_for_event(event_id)
profile_dict = raw_profile_data.to_dict()

# example 2: get related metric data for a specific event. Replace event_id with the ID of the event you are interested in.
event_id = '6e3MwdH2fWm'

raw_metric_data = klaviyo.Events.get_metric_for_event(event_id)
metric_dict = raw_metric_data.to_dict()

File: events/get_related_data_for_event.js
// this JavaScript code is used to get related profile and metric data for a given event.

import { ApiKeySession, EventsApi } from 'klaviyo-api'

// load private API key as an environment variable. Note that this API key needs to have read access to events, profiles, and metrics.
// if using a .env file, ensure to load it using dotenv or similar package. 
const apiKey = process.env.KLAVIYO_API_KEY
if (!apiKey) {
  throw new Error('Please set the KLAVIYO_API_KEY environment variable.')
}

const session = new ApiKeySession(apiKey)
const eventsApi = new EventsApi(session)

// example 1: get related profile data for a specific event. Replace profileEventId with the ID of the event you are interested in.
// note that this is the event's Activity ID rather than Unique ID.
const profileEventId = '6e3MwdH2fWm'

// get profile data related to the event
const profileData = await eventsApi.getProfileForEvent(`${profileEventId}`)

// example 2: get related metric data for a specific event. Replace metricEventId with the ID of the event you are interested in.
const metricEventId = '6e3MwdH2fWm'

// get metric data related to the event
const metricData = await eventsApi.getMetricForEvent(`${metricEventId}`)